crypto/src/secp256k1.rs

```diff
@@ -393,3 +393,14 @@ impl From<Secp256k1PrivateKey> for Secp256k1KeyPair {
         Secp256k1KeyPair { name, secret }
     }
 }
+
+impl Secp256k1Signature {
+    /// Recover public key from signature
+    pub fn recover(&self, hashed_msg: &[u8]) -> Result<Secp256k1PublicKey, signature::Error> {
+        let hashed_msg = rust_secp256k1::Message::from_slice(hashed_msg).unwrap();
```

Comment:
By doing directly the `unwrap` a panic will be raised if the message is invalid - which can happen. I would suggest to refactor this and return an/the error instead when the  result is not an Ok.